PKHeX.Core/Legality/Areas/EncounterArea7.cs

```diff
@@ -8,74 +8,36 @@ namespace PKHeX.Core
     /// </summary>
     public sealed class EncounterArea7 : EncounterArea32
     {
-        protected override IEnumerable<EncounterSlot> GetFilteredSlots(PKM pkm, IEnumerable<EncounterSlot> slots, int minLevel)
+        public override IEnumerable<EncounterSlot> GetMatchingSlots(PKM pkm, IReadOnlyList<EvoCriteria> chain)
         {
-            int species = pkm.Species;
-            int form = pkm.AltForm;
-
             // Edge Case Handling
-            switch (species)
+            switch (pkm.Species)
```

Comment:
This edge case handling might not be necessary anymore due to the cleaned form matching. To be tested later.